libs/clients/directorate-of-immigration/src/lib/directorateOfImmigrationClient.service.ts

```diff
@@ -0,0 +1,608 @@
+          telephone: application.phone,
+          addressCity: application.address,
+        },
+      })
+    } catch (e) {
+      const errorMsg = `Citizenship: Error when trying to create/update applicant to UTL for applicant ${auth.nationalId}`
+      this.logger.error(errorMsg, e)
+      throw new Error(errorMsg)
+    }
+
+    // applicant: post static stata
+    let staticDataId: string
+    try {
+      staticDataId = await this.staticDataApiWithAuth(auth).apiStaticDataPost({
+        staticDataNewModel: {
+          ssn: auth.nationalId,
+          name: application.givenName + ' ' + application.familyName,
+          address: application.address,
+          postalCode: application.postalCode,
+          municipality: application.city,
+          phone: application.phone,
+          nationality: application.citizenshipCode,
+          dateOfDomicileRegistration:
+            application.residenceInIcelandLastChangeDate?.toISOString(),
+          countryOfBirth: application.birthCountry,
+          maritalStatus: application.maritalStatusCode,
+          dateOfMarriage: application.dateOfMaritalStatus?.toISOString(),
+          spouseSSN: application.spouse?.nationalId,
+          spouseName:
+            application.spouse &&
+            application.spouse.givenName + ' ' + application.spouse.familyName,
+          spouseCountryOfBirth: application.spouse?.birthCountry,
+          spouseCitizenship: application.spouse?.citizenshipCode,
+          spouseAddress: application.spouse?.address,
+          spouseAddressMismatchReason:
+            application.spouse?.reasonDifferentAddress,
+          applicantIsChildOfIcelandicCitizen: application.parents.length > 0,
+          parent1SSN: application.parents[0]?.nationalId,
+          parent1Name:
+            application.parents[0] &&
+            application.parents[0]?.givenName +
+              ' ' +
+              application.parents[0]?.familyName,
+          parent2SSN: application.parents[1]?.nationalId,
+          parent2Name:
+            application.parents[1] &&
+            application.parents[1]?.givenName +
+              ' ' +
+              application.parents[1]?.familyName,
+          applicantIsFormerIcelandicCitizen:
+            application.isFormerIcelandicCitizen,
+        },
+      })
+    } catch (e) {
+      const errorMsg = `Citizenship: Error when trying to post static data to UTL for applicant ${auth.nationalId}`
+      this.logger.error(errorMsg, e)
+      throw new Error(errorMsg)
+    }
+
+    // applicant: create application
+    let applicationId: string
+    try {
+      applicationId = await this.applicationApiWithAuth(
+        auth,
+      ).apiApplicationCitizenshipPost({
+        applicationCitizienshipNewModel: {
+          staticDataId,
+        },
+      })
+
+      // clean applicationId and remove double quotes that is added by the openapi generator
+      applicationId = applicationId.replace('"', '').replace('"', '')
```

Comment:
I found this strange so looking at the `clientConfig.json` I see the document 3 content types in response where the first is `text/plain` which the generator is using and hence the response is the plain string. 

Here is a opportunity to guide the service provider on how the openapi document could be improved for easier consumption.

Also you can use regex in `replace` to replace all occurances of a pattern instead of needing to repeat the function.

Reply:
Turned out to be a bug in our openapi generator library. So the content-type definition of the response is ignored. So just unify the replace methods to catch all occurrences using regex.